### DISCUSSION TITLE
keepAlive: true causes the value provided to be nullable

### DISCUSSION THREAD

**Discussion by benji-farquhar** [rrousselGit/riverpod#3834] [Q&A] [answered]
When `keepAlive` is `true`, `ref.read(themeModeNotifierProvider)` is `ThemeMode?`. When `false`, it is `ThemeMode`. Why is that?

```
@Riverpod(keepAlive: true)
class ThemeModeNotifier extends _$ThemeModeNotifier {
  @override
  ThemeMode build() {
    return ref.read(themeModeRepoProvider).fetchLocal();
  }

  void setTheme(ThemeMode themeMode) {
    state = themeMode;
    ref.read(themeModeRepoProvider).saveLocal(themeMode);
  }
}
```

**Comment by AhmedLSayed9** [comment, marked as answer]
It can't be `ThemeMode?` as your state is `ThemeMode`. Maybe you're using stale generated code and need to re-run build runner.
Also, You should use `ref.watch` at build method.

**Comment by benji-farquhar** [comment]
@AhmedLSayed9 You're right, it was stale 😅 Thank you

**Reply by AhmedLSayed9** [reply]
Don't forget to mark the discussion as answered 😛
